zetaclient/chains/evm/signer/v2_sign.go

```diff
@@ -29,19 +29,17 @@ func (signer *Signer) signGatewayExecute(
 
 	var data []byte
 
+	// only set sender if it's authenticated call
```

Comment:
Can we precise here the smart contract will route logic to arbitrary call when not set?  This appears a bit like an implicit behavior here